core/class/jMQTT.class.php

```diff
@@ -689,8 +689,12 @@ public function postRemove() {
     public static function health() {
         $return = array();
         foreach(self::getBrokers() as $broker) {
+            // if(!$broker->getIsEnable() || $broker->getMqttClientState() == jMQTTBase::MQTTCLIENT_POK || $broker->getMqttClientState() == jMQTTBase::MQTTCLIENT_NOK)
+                // continue;
             $mosqHost = $broker->getConf(self::CONF_KEY_MQTT_ADDRESS);
             $mosqPort = $broker->getConf(self::CONF_KEY_MQTT_PORT);
+            if ($mosqPort == '')
+                $mosqPort = (boolval($broker->getConf(self::CONF_KEY_MQTT_TLS))) ? 8883 : 1883;
```

Comment:
Woule be better to handle default port using jMQTT::getDefaultConfiguration()

Reply:
Ok but there a some implications:
- _jMQTT::getDefaultConfiguration_ cannot be static anymore, as _CONF_KEY_MQTT_PORT_ **default** value depends on _CONF_KEY_MQTT_TLS_ **current** value.
- _CONF_KEY_MQTT_PORT_ needs to be outside of _$defValues_ (in _jMQTT::getDefaultConfiguration_), because is some (frequent) cases _CONF_KEY_MQTT_TLS_ needs to exist before _CONF_KEY_MQTT_PORT_, as it's unset it creates a recution loop while trying to create _$defValues_ array.

See my modification of the PR.